src/peripheral/dwt.rs

```diff
@@ -356,12 +356,23 @@ pub struct ComparatorAddressSettings {
     pub access_type: AccessType,
 }
 
+/// Settings for cycle count matching
+#[derive(Debug, Eq, PartialEq, Copy, Clone)]
+pub struct CycleCountSettings {
```

Comment:
Could you add support for the two other cycle count comparison functions: PC value packet generation (`0b0001`) and `CMPMATCH[N]` event generation (`0b1000`)?

Reply:
If not, please refactor in some `#[non_exhaustive]` enum that's used to select the watchpoint debug event function.

Reply:
Added and also extended the existing code for the new `EmitOption`'s